failpoints.go

```diff
@@ -186,11 +186,7 @@ func (fps *Failpoints) Eval(failpath string) (Value, error) {
 		return nil, ErrNotExist
 	}
 
-	val, err := fp.Eval()
-	if err != nil {
-		return nil, fmt.Errorf("%v on %s", err, failpath)
-	}
-	return val, nil
+	return fp.Eval()
```

Comment:
maybe use `fmt.Errorf("%w on %s", err, failpath)` or use `errors.Annotatef`?

then in `EvalContext` check using `errors.Unwrap` (for built-in errors) / `errors.Cause` (for pingcap/errors).

Reply:
`errors.Unwrap` is not supported by the old go versions. `errors.Cause` works, however, the error handling here is too simple to introduce the `pingcap/errors`.

I really prefer the solution in this PR, but if you insist, I can change to your way.

Reply:
The problem of removing the `fmt.Errorf` is that the `failpath` is lost when using `(*Failpoints).Eval` directly.

I think it is safe to upgrade the CI's Go version to 1.13+, as the latest version is 1.14, and all TiDB projects are targeting 1.13+.

(BTW in the actual implementation we should use `errors.Is`, not `errors.Unwrap`)

Reply:
`failpath`  is not lost, it is supplied by the user, the user should handle the error and failpath together if he/she would like to.

It is not just a problem of CI, some other users may still use older go versions, there is no more benefit to just give up these users.

Reply:
@kennytm Do you have any more suggestions? Or are you still worry about the `failpath`?

Considering that we have many methods of type `Failpoints`, like `Status`, `Enable`, `Disable`, `List`... They all return the original error directly. I think it was a design mistake to return an error with the `failpath`. Now it's time to fix it. I am blocked on this PR for a long time, looking forward to your reply.

Reply:
All filtered errors are defined in the package `failpoint`, I think we can define a customized error type to attach the extra information.

Reply:
@lonng OK, going to try that.

Reply:
@lonng Use the FpError to indicates the internal errors